jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleProjectProperties.java

```diff
@@ -148,20 +141,87 @@ private static boolean isProgressFooterEnabled(Project project) {
   private final Project project;
   private final SingleThreadedExecutor singleThreadedExecutor = new SingleThreadedExecutor();
   private final EventHandlers eventHandlers;
-  private final JavaLayerConfigurations javaLayerConfigurations;
+  private final Logger logger;
+  private final AbsoluteUnixPath appRoot;
 
   @VisibleForTesting
-  GradleProjectProperties(
-      Project project, Logger logger, JavaLayerConfigurations javaLayerConfigurations) {
+  GradleProjectProperties(Project project, Logger logger, AbsoluteUnixPath appRoot) {
     this.project = project;
-    this.javaLayerConfigurations = javaLayerConfigurations;
+    this.logger = logger;
+    this.appRoot = appRoot;
 
     eventHandlers = makeEventHandlers(project, logger, singleThreadedExecutor);
   }
 
   @Override
-  public JavaLayerConfigurations getJavaLayerConfigurations() {
-    return javaLayerConfigurations;
+  public JibContainerBuilder getContainerBuilderWithLayers(RegistryImage baseImage) {
+    try {
+      if (isWarProject()) {
+        logger.info("WAR project identified, creating WAR image: " + project.getDisplayName());
+        Path explodedWarPath = GradleProjectProperties.getExplodedWarDirectory(project);
+        return JavaContainerBuilderHelper.fromExplodedWar(baseImage, explodedWarPath, appRoot);
+      }
+
+      JavaPluginConvention javaPluginConvention =
+          project.getConvention().getPlugin(JavaPluginConvention.class);
+      SourceSet mainSourceSet =
+          javaPluginConvention.getSourceSets().getByName(MAIN_SOURCE_SET_NAME);
+
+      FileCollection classesOutputDirectories =
+          mainSourceSet.getOutput().getClassesDirs().filter(File::exists);
+      Path resourcesOutputDirectory = mainSourceSet.getOutput().getResourcesDir().toPath();
+      FileCollection allFiles = mainSourceSet.getRuntimeClasspath();
+      FileCollection dependencyFiles =
+          allFiles
+              .minus(classesOutputDirectories)
+              .filter(file -> !file.toPath().equals(resourcesOutputDirectory));
+
+      JavaContainerBuilder javaContainerBuilder =
+          JavaContainerBuilder.from(baseImage).setAppRoot(appRoot);
+
+      // Adds resource files
+      if (Files.exists(resourcesOutputDirectory)) {
+        javaContainerBuilder.addResources(resourcesOutputDirectory);
+      }
+
+      // Adds class files
+      for (File classesOutputDirectory : classesOutputDirectories.filter(File::exists)) {
+        if (classesOutputDirectory.exists()) {
+          javaContainerBuilder.addClasses(classesOutputDirectory.toPath());
+        }
+      }
+      if (classesOutputDirectories.filter(File::exists).isEmpty()) {
```

Comment:
Perhaps just `classesOutputDirectories.filter.isEmpty()` because it is already filtered?